torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -264,6 +264,59 @@ std::shared_ptr<SugaredValue> ClassValue::attr(
   }
   return std::make_shared<ClassNewMethod>(type_);
 }
+
+std::shared_ptr<SugaredValue> NamedTupleConstructor::call(
+    const SourceRange& loc,
+    Function& m,
+    at::ArrayRef<NamedValue> inputs,
+    at::ArrayRef<NamedValue> attributes,
+    size_t n_binders) {
+  auto nargs = type_->containedTypes().size();
+  if (inputs.size() != nargs) {
+    throw ErrorReport(loc) << "Constructor expected " << nargs << "arguments "
+                           << "but got " << inputs.size();
+  }
+  // TODO: Implement construction with kwargs
+  if (attributes.size() != 0) {
+    throw ErrorReport(loc) << "NamedTuple construction with kwargs is currently"
+                           << " not supported";
+  }
+
+  auto& g = *m.graph();
+
+  // Check that the types of the inputs passed into the NamedTuple constructor
+  // match the expected types
+  std::vector<Argument> arguments;
+  for (size_t i = 0; i < nargs; ++i) {
+    arguments.emplace_back(
+        /*name=*/type_->names()[i],
+        /*type=*/type_->containedTypes()[i],
+        /*N=*/i);
+  }
+
+  TORCH_INTERNAL_ASSERT(type_->unqualName());
+  FunctionSchema schema(
+      /*name=*/type_->unqualName().value(),
+      /*overload_name=*/"",
+      /*arguments=*/arguments,
+      /*returns=*/{});
+
+  matchSchema(schema, loc, g, inputs, attributes);
```

Comment:
Look at other uses of matchSchema. This needs to take the _output_ of match schema and pass it into tuple construct. Otherwise, this introduces type errors (implicit conversions were run and discarded). Also this would trivially handle keyword arguments. Take a look at the other places where matchSchema is used.